### PR TITLE
React bootstrap upgrade to v1, migrate to react-beautiful-dnd

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Upgraded to `bootstrap` v4, `react-bootstrap` v1, and removed `react-smooth-dnd` in favour of `react-beautiful-dnd`.
Note that there will be some visual changes happening here as a result of the change to Bootstrap v4, so I'd like UX to have a look as well before this is merged.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22255

#### Release Note
```release-note
Migrated to Bootstrap v4 and refreshed interface.
Migrated to react-beautiful-dnd instead of react-smooth-dnd for a cleaner experience.
```

### CODE REVIEW

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Looks great @devinbinnie. Nothing big from me, but I did have a couple requests:

- For the add/edit/remove modals, can we make sure the `Cancel` button is treated consistently? I would use the style used on the Cancel button in the Remove confirmation modal.
Screen Shot 2021-06-28 at 10 46 46 AM
Screen Shot 2021-06-28 at 10 47 05 AM

Also, are we able to update the button blue to be the same blue as the default theme button color (#166DE0)?

**Review by matthewbirtch — APPROVED** [review]

File: src/renderer/components/SettingsPage.tsx [hunk 1]
```diff
@@ -468,85 +462,95 @@ export default class SettingsPage extends React.PureComponent<Record<string, nev
...
-                    <HelpBlock>
+                    <FormText>
                         {'If enabled, the app starts automatically when you log in to your machine.'}
-                    </HelpBlock>
-                </Checkbox>);
+                    </FormText>
+                </FormCheck>);
         }
 
+        console.log(this.state.useSpellChecker);
```
Comment (deanwhillier) [inline src/renderer/components/SettingsPage.tsx:481]: I'm imagining this was just missed? 😛

File: src/renderer/components/SettingsPage.tsx [hunk 2]
```diff
@@ -604,30 +609,33 @@ export default class SettingsPage extends React.PureComponent<Record<string, nev
...
-                    <Radio
+                    <FormCheck
+                        type='radio'
                         inline={true}
                         name='trayIconTheme'
                         value='dark'
                         defaultChecked={this.state.trayIconTheme === 'dark'}
                         onChange={() => this.handleChangeTrayIconTheme('dark')}
-                    >{'Dark'}</Radio>
+                        label={'Light'}
```
Comment (deanwhillier) [inline src/renderer/components/SettingsPage.tsx:629]: This was previously 'Dark' before, is the change intentional?

**Review by deanwhillier — CHANGES_REQUESTED** [review]
Just a couple things below.

File: src/renderer/components/SettingsPage.tsx (hunk 1 above)
Comment (devinbinnie) [reply src/renderer/components/SettingsPage.tsx:481]: Yep lol

File: src/renderer/components/SettingsPage.tsx (hunk 2 above)
Comment (devinbinnie) [reply src/renderer/components/SettingsPage.tsx:629]: Oops, nope

**Review by Willyfrog — APPROVED** [review]
LGTM! thanks for upgrading react-bootstrap!

**Review by deanwhillier — APPROVED** [review]

**Pull request merged**